flink-ml-lib/src/main/java/org/apache/flink/ml/classification/linear/LogisticGradient.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.classification.linear;
+
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.api.java.tuple.Tuple3;
+import org.apache.flink.ml.linalg.BLAS;
+
+import java.io.Serializable;
+
+/**
+ * Utility class to compute gradient and loss for logistic loss. For details, see
+ * http://mlwiki.org/index.php/Logistic_Regression.
+ */
+public class LogisticGradient implements Serializable {
+
+    /** L2 regularization term. */
+    private final double l2;
+
+    public LogisticGradient(double l2) {
+        this.l2 = l2;
+    }
+
+    /**
+     * Computes loss and sum of weight on a set of samples.
+     *
+     * @param labeledData A sample set of train data.
+     * @param coefficient The model parameters.
+     * @return Loss and weight sum of the input data.
+     */
+    public final Tuple2<Double, Double> computeLoss(
```

Comment:
In the existing Flink codebase, if the input is always `List`, we use `List` as the input type instead of using `Iterable`. Could we use  the same approach here for consistency?